packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2605,6 +2605,12 @@ class RenderTransform extends RenderProxyBox {
       if (filterQuality == null) {
         final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
         if (childOffset == null) {
+          // if either the x or y is scaled by `0.0` nothing will be drawn anyway,
+          // so short-circuit and avoid more work.
+          if (transform[0] == 0.0 || transform[5] == 0.0) {
```

Comment:
Ahh, then I suspect this change didn't fail any tests because we end up with extremely small values for these when using Transform.rotate. If I had used some epsilon that might have made the error more obvious.

Reply:
Rotating using radians is always going to be hit or miss as "90 degrees" plus the bit errors in the trig functions means you never really get a true quadrant rotation. I worked around that in Java by [testing sin/cos for +/-1](https://github.com/openjdk/jdk/blob/351560414d7ddc0694126ab184bdb78be604e51f/src/java.desktop/share/classes/java/awt/geom/AffineTransform.java#L1430) and then forcing the other parameter to 0 (actually calling a different, faster quadrant rotation method). (It turns out that sin/cos produce a "1.0" answer over a wider range of inputs than their counterpart produces a 0. And if one of them is claiming "1.0" then there is no point in maintaining tiny precision in the other.)

My suspicion is that we probably didn't have any tests for rotated cases?